src/Lib.P2P.Tests/MultiAddressExtensionsTest.cs

```diff
@@ -0,0 +1,94 @@
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Linq;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MultiFormats;
+
+namespace Lib.P2P.Tests
+{
+    [TestClass]
+    public class MultiAddressExtensionsTest
+    {
+        [TestMethod]
+        public void Cloning()
+        {
+            var a = new MultiAddress("/dns/libp2p.io/tcp/5001");
+            var b = a.Clone();
+            Assert.AreEqual(a, b);
+            Assert.AreNotSame(a.Protocols, b.Protocols);
+        }
+
+        [TestMethod]
+        public async Task Resolving()
+        {
+            var local = new MultiAddress("/ip4/127.0.0.1/tcp/5001");
+            var r0 = await local.ResolveAsync();
+            Assert.AreEqual(1, r0.Count);
+            Assert.AreEqual(local, r0[0]);
+        }
+
+        [TestMethod]
+        public async Task Resolving_Dns()
+        {
+            var dns = await new MultiAddress("/dns/libp2p.io/tcp/5001").ResolveAsync();
+            Assert.AreNotEqual(0, dns.Count);
+            var dns4 = await new MultiAddress("/dns4/libp2p.io/tcp/5001").ResolveAsync();
+            var dns6 = await new MultiAddress("/dns6/libp2p.io/tcp/5001").ResolveAsync();
+            Assert.AreEqual(dns.Count, dns4.Count + dns6.Count);
+        }
+
+        [TestMethod]
+        public async Task Resolving_HTTP()
+        {
+            var r = await new MultiAddress("/ip4/127.0.0.1/http").ResolveAsync();
+            Assert.AreEqual("/ip4/127.0.0.1/http/tcp/80", r.First());
+
+            r = await new MultiAddress("/ip4/127.0.0.1/http/tcp/8080").ResolveAsync();
+            Assert.AreEqual("/ip4/127.0.0.1/http/tcp/8080", r.First());
+        }
+
+        [TestMethod]
+        public async Task Resolving_HTTPS()
+        {
+            var r = await new MultiAddress("/ip4/127.0.0.1/https").ResolveAsync();
+            Assert.AreEqual("/ip4/127.0.0.1/https/tcp/443", r.First());
+
+            r = await new MultiAddress("/ip4/127.0.0.1/https/tcp/4433").ResolveAsync();
+            Assert.AreEqual("/ip4/127.0.0.1/https/tcp/4433", r.First());
+        }
+
+        [TestMethod]
+        public void Resolving_Unknown()
+        {
+            ExceptionAssert.Throws<SocketException>(() =>
+            {
+                var _ = new MultiAddress("/dns/does.not.exist/tcp/5001")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)